File: DS Week 1&2/trials/stack.js
class ArrayStack{
    constructor(){
        this.value=[]
    }
    push(element){
        this.value.unshift(element)
    }
    pop(){
        this.value.shift()
    }
    print(){
        console.log(this.value.toString());
    }
}
const stack=new ArrayStack()
stack.push(1)
stack.push(2)
stack.push(3)
stack.print()
class Node{
    constructor(element){
        this.value=element
        
    }
    next=null
}
class LinkedListedStack{
    head=null
    push(element){
        const newNode=new Node(element)
        if(this.head!=null){
            newNode.next=this.head
        }
        this.head=newNode
    }
    pop(){
        if(this.head==null){
            newNode.next=this.head
        }else{
            this.head=this.head.next
        }
    }
    print (){
        let current=this.head
        while(current!=null){
            console.log(current.value)
            current=current.next
        }
    }
}
const linkedStack=new LinkedListedStack()
linkedStack.push(10)
linkedStack.push(20)
linkedStack.push(30)
linkedStack.pop()
linkedStack.print()

File: DS Week 1&2/Stack/Stack.java
public class Stack {
    public static void main(String[] args) {
        Stack stack1=new Stack();
        stack1.push(10);
        stack1.push(20);
        stack1.push(30);
        stack1.pop();
        stack1.display();
        
    }   

    class Node{
        int data;
        Node next;
        Node (int data){
            this.data=data;
        }
    }
    Node top;
    void display(){
        if(top==null){
            System.out.println("emty");
            return;
        }
        Node temp=top;
        while(temp!=null){
            System.out.println(temp.data);
            temp=temp.next;
        }
    }
    void push(int data){
        Node newNode=new Node(data);
        if(top==null){
            top=newNode;
        }else{
            newNode.next=top;
            top=newNode;
        }
        
    }
    void pop(){
        if(top==null){
            System.out.println("Stack underflow");
            return;
        }else{
            top=top.next;
        }
        
    }
}
